src/mysql/mod.rs

```diff
@@ -0,0 +1,120 @@
+use std::{collections::HashMap, time::Duration};
+
+use testcontainers::{core::WaitFor, Image};
+
+const NAME: &str = "mysql";
+const TAG: &str = "8.1";
+
+#[derive(Debug)]
+pub struct Mysql {
+    env_vars: HashMap<String, String>,
+}
+
+impl Default for Mysql {
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+        env_vars.insert("MYSQL_DATABASE".to_owned(), "mysql".to_owned());
+        env_vars.insert("MYSQL_ALLOW_EMPTY_PASSWORD".into(), "yes".into());
+
+        Self { env_vars }
+    }
+}
+
+impl Image for Mysql {
+    type Args = ();
+
+    fn name(&self) -> String {
+        NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        TAG.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![
+            WaitFor::message_on_stderr("/usr/sbin/mysqld: ready for connections."),
+            WaitFor::Duration {
+                length: Duration::new(10, 0),
+            },
```

Comment:
Do we really need it? Probably one more message can indicate it's ready?
If so, consider using https://docs.rs/testcontainers/0.15.0/testcontainers/core/enum.WaitFor.html#method.seconds

Reply:
Yeah, I don't like it either. Apparently mysql container restarts once due to (my best guess) password setup or some other janitorial work they do in the entrypoint. We could detect if at least the ready for connections message is output twice instead of the WaitFor::Duration I was in a rush so this is the fastest thing I managed to come up with this morning. I can investigate a bit further

Reply:
I thought about adding a healthcheck something I have done in other [projects](https://github.com/passbolt/passbolt_docker/blob/58b0ea821eec8bb55e0e3e1b25ca8b8e819b737e/spec/docker_runtime/runtime_spec.rb#L21) but I think I would need to open a PR on the testcontainers repo as I don't see any method to define it, right?

Reply:
In the end I have created a PR on testcontainers-rs to support the addition of a health-cmd that would allow us to define a custom healthcheck and so not relying on container output or duration. If that gets merged latest changes in this branch will work fine (updating cargo version for testcontainers-rs). https://github.com/testcontainers/testcontainers-rs/pull/529

Reply:
I'd not expect the PR to be reviewed & released quickly (last release has took ~ a year, due to lack of maintainers).
So would suggest to implement something working with current version and then we can update the implementation.

Reply:
Alright! Sounds good. I have just pushed some changes to check on another condition on stderr. As the container always binds a port in the scenario of the testcontainers it makes sense. This might be good enough till a new testcontainers version. Let me know what you think and sorry for all the back and forth!

Reply:
Thanks for all your effort to add one more image!